PhysicsTools/NanoAOD/plugins/NanoAODDQM.cc

```diff
@@ -195,9 +215,10 @@ class NanoAODDQM : public DQMEDAnalyzer {
     std::vector<SelGroupConfig> selGroups;
   };
   std::map<std::string, GroupConfig> groups_;
+  edm::GetterOfProducts<FlatTable> getterOfProducts_;
 };
 
-NanoAODDQM::NanoAODDQM(const edm::ParameterSet &iConfig) {
+NanoAODDQM::NanoAODDQM(const edm::ParameterSet &iConfig) : getterOfProducts_(edm::ProcessMatch("NANO"), this) {
```

Comment:
being curious : does this mean the process name has to be NANO, otherwise this won't work ?

Reply:
I suppose... From the twiki, using `"*"` should work, this might be safer indeed. Perhaps @wddgit can chime in?

Reply:
Yes, if you use ProcessMatch with "NANO" it requires the process name to be "NANO". If instead you use "*" as a wildcard any process name would be allowed and then it behaves the same as getManyByType. That should work fine.

The predicate classes are very simple. We could also add new one that always returns true. You could look at the pattern here:

https://cmssdt.cern.ch/lxr/source/FWCore/Framework/interface/ProcessMatch.h#0004

It is just a functor that takes a BranchDescription as an argument and returns a bool. You might even be able to use a simple lambda. I don't think those existed yet when we originally wrote GetterOfProducts. I never tried that.

Reply:
Ok, thanks. @vlimant do you see a good reason for accepting any process name?

Reply:
if `"*"` is allowed, let's use that instead of hardcoding that the process name has to be. A MINI+NANO workflow would likely be call PAT for example

Reply:
good point